tests/test_nodes.py

```diff
@@ -659,7 +660,22 @@ def test_decay_queue(self):
             "temperature": ((5 * 10) + (15 * 1)) / 6,
         }
         self.assertDictAlmostEqual(d4, tank.storage, 15)
-
+    
+    def test_node_overrides(self):
+        node = Node(name="", data_input_dict={("temperature", 1): 15})
+        # check the format of dict
+        new_data_input_dict = {("temperature", 1): 10,
+                               ("temperature", 2): 20,
+                               }
+        node.apply_overrides({'data_input_dict': new_data_input_dict})
+        self.assertDictEqual(node.data_input_dict, new_data_input_dict)
+        # check the format of str
+        new_data_input_dict = "example_data_input_dict.csv.gz"
+        node.apply_overrides({'data_input_dict': new_data_input_dict})
+        from wsimod.orchestration.model import read_csv
+        new_data_input_dict = read_csv(new_data_input_dict)
+        self.assertDictEqual(node.data_input_dict, new_data_input_dict)
+        print(dict(list(node.data_input_dict.items())[:5]))
```

Comment:
I think would be tidier with `tempfile` (and you can also delete the example data dict file):

```suggestion
        from pathlib import Path
        import tempfile
        with tempfile.TemporaryDirectory as temp_dir:
                fid = Path(temp_dir) / "example_data_input_dict.csv"
                pd.DataFrame(new_data_input_dict).to_csv(fid)
                node.apply_overrides({'data_input_dict': fid})
                self.assertDictEqual(node.data_input_dict, new_data_input_dict)
```

Reply:
an error raises
```
    with tempfile.TemporaryDirectory as temp_dir:
AttributeError: __enter__
```

Reply:
oops needed `tempfile.TemporaryDirectory()` - but I'm happy to ignore it as is only a test